Sources/Units/Unit/Fraction.swift

```diff
@@ -0,0 +1,163 @@
+        Self(numerator: (lhs.numerator * rhs.denominator) - (rhs.numerator * lhs.denominator), denominator: lhs.denominator * rhs.denominator)
+    }
+}
+extension Fraction {
+    public static func * (lhs: Self, rhs: Int) -> Self {
+        lhs * Self(integerLiteral: rhs)
+    }
+
+    public static func / (lhs: Self, rhs: Int) -> Self {
+        lhs / Self(integerLiteral: rhs)
+    }
+
+    public static func * (lhs: Int, rhs: Self) -> Self {
+        Self(integerLiteral: lhs) * rhs
+    }
+
+    public static func / (lhs: Int, rhs: Self) -> Self {
+        Self(integerLiteral: lhs) * rhs
+    }
+
+    public static func + (lhs: Self, rhs: Int) -> Self {
+        lhs + Self(integerLiteral: rhs)
+    }
+
+    public static func - (lhs: Self, rhs: Int) -> Self {
+        lhs - Self(integerLiteral: rhs)
+    }
+
+    public static func + (lhs: Int, rhs: Self) -> Self {
+        Self(integerLiteral: lhs) + rhs
+    }
+
+    public static func - (lhs: Int, rhs: Self) -> Self {
+        Self(integerLiteral: lhs) + rhs
+    }
+}
+
+extension Fraction: ExpressibleByIntegerLiteral {
+    public typealias IntegerLiteralType = Int
+
+    public init(integerLiteral value: Int) {
+        self.init(numerator: value, denominator: 1)
+    }
+}
+
+extension Fraction: SignedNumeric {
+
+    public init?<T>(exactly source: T) where T : BinaryInteger {
+        self.init(integerLiteral: Int(source))
+    }
+
+    public static func *= (lhs: inout Fraction, rhs: Fraction) {
+        lhs = lhs * rhs
+    }
+
+    public var magnitude: Fraction {
+        Self(numerator: abs(numerator), denominator: abs(denominator))
+    }
+    
+    public typealias Magnitude = Self
+
+}
+
+extension Fraction {
+    var asDouble: Double {
+        Double(numerator) / Double(denominator)
+    }
+}
+
+extension Fraction: Comparable {
+    public static func < (lhs: Fraction, rhs: Fraction) -> Bool {
+        return lhs.numerator < rhs.numerator
+    }
+}
+
+extension Fraction: LosslessStringConvertible {
+    public init?(_ description: String) {
+        if 
+            description.first == "(",
+            description.last == ")"
+        {
+            let parts = description.dropFirst().dropLast().split(separator: "|").compactMap({ Int(String($0)) })
+            guard
+                parts.count == 2,
+                let numerator = parts.first,
+                let denominator = parts.last
+            else {
+                return nil
+            }
+            self.init(numerator: numerator, denominator: denominator)
+        } else if let number = Int(description) {
+            self.init(integerLiteral: number)
+        } else {
+            return nil
+        }
+    }
+
+    public var description: String {
+        if denominator == 1 {
+            "\(numerator)"
+        } else {
+            "(\(numerator)|\(denominator))"
+        }
+    }
+}
+
+extension SignedInteger {
+    func over<T: SignedInteger>(_ denominator: T) -> Fraction {
```

Comment:
Were there type inference issues with using `/`, like `5.measured(in: .meter.pow(1/2))`?

Reply:
Yeah. That was what I originally tried.
It was ambiguous to the compiler whether `1 / 2` meant `Double / Double` or `Fraction / Double` or `Fraction / Fraction` etc because `Fraction` conforms to ExpressibleAsIntegerLiteral.

Reply:
On that note, how do you feel about `|` as the operator? It wouldn't be ambiguous.

You could do `2|5` etc

Reply:
I went ahead and did this - lmk if you like/dislike.